lib/claim_letters/claim_letter_downloader.rb

```diff
@@ -68,12 +71,13 @@ def format_letter_data(docs)
       letters.sort_by { |d| d[:received_at] }.reverse
     end
 
-    def verify_letter_in_folder(document_id)
+    def get_letter_details(document_id)
       letters = get_letters
+      letters.select { |d| d[:document_id] == document_id }
+    end
 
-      raise Common::Exceptions::Unauthorized unless letters.any? do |document|
-        document[:document_id] == document_id
-      end
+    def filename_with_date(filename, filedate)
+      "#{filename}-#{filedate.year}-#{filedate.month}-#{filedate.day}.pdf"
```

Comment:
What happens (also, _could_ this happen?) if a letter either:
1. does not have a `filedate` (null pointer exception, I'm guessing) or
2. does not have a valid year, month, and/or day?

Another way of asking this is: are we always guaranteed to have a present and valid `filedate`?

If any of these cases are possible, we should probably handle them gracefully and add tests to cover.

Reply:
Any thoughts on this @jerekshoe ? Do we think received_at would ever be blank? It seems unlikely to me.

Reply:
Willing to handle just to be safe, regardless.

Reply:
It shouldn't ever be blank, but just to be safe, have you verified what is returned if it is blank?

Reply:
Another way this question could be phrased is: is the `filedate` input by humans? does the input system allow blank `filedate`s? We might not know the answers, but it may help to think about it differently.

Reply:
No, the filedate is something that is system generated

Reply:
The `received_at` date is a Date object, so presumably could be nil. When I test the nil version in our test data, it currently triggers a `comparison of Date with nil failed` error at `claim_letter_downloader.rb:68:in `sort_by'`

Reply:
Cool beans. If `received_at` is system generated, there's a low likelihood of it being nil. To avoid spending more time on this unlikely case, we could put a `TODO: (rare) Handle nil received_at` comment above line 68. Thoughts?

In any event, thanks for checking - y'all helped me learn more about the underlying systems.